base/linalg/triangular.jl

```diff
@@ -1894,15 +1884,20 @@ function sqrtm{T,realmatrix}(A::UpperTriangular{T},::Type{Val{realmatrix}})
     if realmatrix
         TT = typeof(sqrt(zero(T)))
     else
-        TT = typeof(sqrt(complex(-one(T))))
+        TT = typeof(sqrt(complex(zero(T))))
     end
-    n = checksquare(A)
+    n = Base.LinAlg.checksquare(A)
     R = zeros(TT, n, n)
-    for j = 1:n
-        R[j,j] = realmatrix ? sqrt(A[j,j]) : sqrt(complex(A[j,j]))
-        for i = j-1:-1:1
-            r = floop(A[i,j],R,i,j)
-            r==0 || (R[i,j] = r / (R[i,i] + R[j,j]))
+    @inbounds begin
+        for j = 1:n
+            R[j,j] = realmatrix ? sqrt(A[j,j]) : sqrt(complex(A[j,j]))
+            for i = j-1:-1:1
+                r = A[i,j] + zero(TT)
```

Comment:
To be clear, do you suggest replacing

    TT = realmatrix ? typeof(sqrt(zero(T))) : typeof(sqrt(complex(zero(T))))
with

    TA = realmatrix ? float(T) : Complex{float(T)}
? (And then later using `r::TA` or `r::TT` in either case) Sounds good to me, much clearer :+1: